public/js/particle_manager.js

```diff
@@ -0,0 +1,39 @@
+import {Particle} from "./particle.js";
+import {random, randomChoice} from "./util.js";
+
+export class ParticleManager {
+
+    constructor(options) {
```

Comment:
Fulfils requirements 2/3 - delegate to manager to handle responsibility of particle life cycle rather than polluting engine class.